### PR TITLE
Confusing error message from volume.create when datastore is not writ…

### PR DESCRIPTION
**Pull request by BaluDontu**
Handled both scenarios as outlined below:

1. datastore is not writable
2. permission denied to create dockvols directory in datastore.

This is extension to #677 and is part of fix for #568.

osfs-mkdir -n returns different error codes when it is not able to create the dockvols directory in /vmfs/volumes/<datastore-name>. It returns error codes 13 and 30 for scenarios 1 and 2 described above. These error codes are used to give a proper errorMsg to the user.

### CODE REVIEW

**Comment by BaluDontu** [comment]
@pdhamdhere : Can you please review it.

File: esx_service/vmdk_ops.py [hunk 1]
```diff
@@ -475,24 +475,33 @@ def get_vol_path(datastore, tenant_name=None):
...
 
     if not os.path.isdir(dock_vol_path):
         # The osfs tools are usable for DOCK_VOLS_DIR on all datastores
         cmd = "{0} {1}".format(OSFS_MKDIR_CMD, dock_vol_path)
         rc, out = RunCommand(cmd)
-        if rc != 0:
-            logging.warning("Failed to create %s", dock_vol_path)
-            return None
+        errMsg = None
+        if rc == 13:
```
Comment (msterin) [inline esx_service/vmdk_ops.py]: Please do not use literals like 13 or 30 directly. Define ERR_OSFS_MKDIR_<WATEVER>=13 and use here.

File: esx_service/vmdk_ops.py [hunk 2]
```diff
@@ -475,24 +475,33 @@ def get_vol_path(datastore, tenant_name=None):
...
+            errMsg = "{0} is not writable and hence ./dockvol creation failed".format(datastore)
```
Comment (msterin) [inline esx_service/vmdk_ops.py]: nit: A better message IMO would be '{0} creation failed - {1} not writeable).format('dockvols', datastore' (failure first, reasons next)

File: esx_service/vmdk_ops.py [hunk 3]
```diff
@@ -475,24 +475,33 @@ def get_vol_path(datastore, tenant_name=None):
...
+        elif rc == 30:
+            errMsg = "Permission denied to create ./dockvol on {}".format(datastore)
```
Comment (msterin) [inline esx_service/vmdk_ops.py]: nit: I think its "dockvols" not "dockvol". Maybe it's time to have a global var for this ?

File: esx_service/vmdk_ops.py [hunk 4]
```diff
@@ -475,24 +475,33 @@ def get_vol_path(datastore, tenant_name=None):
...
+        else:
+            logging.warning("Failed to create %s", path)
+            errMsg = "Failed to initialize volume path {0}".format(path)
```
Comment (msterin) [inline esx_service/vmdk_ops.py]: different logging and errmsg strings looks strange. Can we use the same string ?

**Pull request closed without merging**